### PR TITLE
tests: add IDF test

### PR DESCRIPTION
**Pull request by zimbatm**
~Broken PR that shows that tests are failing with IFD~

This shows how to use yarn2nix with IFD

### CODE REVIEW

File: run-tests.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+: "${NIXPKGS_CHANNEL:=nixpkgs-unstable}"
+
+nix_options=(
+  --no-out-link
+  #--option allow-import-from-derivation false
```
Comment (moretea) [inline run-tests.sh]: Maybe remove disabled options?

File: run-tests.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  #--option restrict-eval true
+  --option sandbox true
+  --show-trace
+  -I "nixpkgs=channel:$NIXPKGS_CHANNEL"
+)
+
+nix-build tests "${nix_options[@]}"
```
Comment (moretea) [inline run-tests.sh]: We probably want to explicitly set`--option allow-import-from-derivation true` here.

Comment (moretea) [reply run-tests.sh]: Could you also add a comment explaining that this tests yarn with and without IFD?

**Review by moretea — APPROVED** [review]
Great :heart: 

Could you fix the two small nits before merging?

File: run-tests.sh (hunk 1 above)
Comment (zimbatm) [reply run-tests.sh]: Fixed in ffa06ac84a5158cf97609f6c5249cdf0dc8b36c1

File: run-tests.sh (hunk 2 above)
Comment (zimbatm) [reply run-tests.sh]: Good call, fixed in ffa06ac84a5158cf97609f6c5249cdf0dc8b36c1

**Pull request merged**